File: 0001/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const obj: Record<number, number> = {};
  for (let i: number = 0; i < nums.length; i++) {
    let complement: number = target - nums[i];

    if (complement in obj) {
      return [obj[complement], i];
    }
    obj[nums[i]] = i;
  }
  return [];
}

File: 0001/solution.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> cache = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (cache.containsKey(diff)) {
                return new int[]{cache.get(diff), i};
            }
            cache.put(nums[i], i);
        }
        return new int []{};
    }
}